apps/credential-from-input-descriptor/src/modules/converter/dtos/input-descriptor-to-credential-response.dto.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2021, 2022 Energy Web Foundation
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export class InputDescriptorToCredentialResponseDto {
+  '@context'?: (string | Record<string, unknown>)[];
+  credentialSubject?: (string | Record<string, unknown>)[];
+  id?: (string | Record<string, unknown>)[];
+  issuanceDate?: (string | Record<string, unknown>)[];
+  type?: (string | Record<string, unknown>)[];
```

Comment:
Would it be possible to use the `Credential` interface from ew-did-registry as follows? https://github.com/energywebfoundation/ew-did-registry/blob/1290906c8906655e99528ae64ebebe1dea41b5a2/packages/credentials-interface/src/credentials.types.ts#L19

Even if not possible, I think that it is better to have `credential` as a property of the response instead of have the credential properties directly in the DTO. This is because maybe we will want to return something other than the `credential` in the future and having a `credential` makes it easier to do this without it being a breaking change

```suggestion
  credential: Credential<string | Record<string, unknown>>
```

Reply:
This change causes the following error:
```text
src/modules/converter/dtos/input-descriptor-to-credential-response.dto.ts:21:26 - error TS2344: Type 'string | Record<string, unknown>' does not satisfy the constraint 'ICredentialSubject'.
  Type 'string' is not assignable to type 'ICredentialSubject'.

21   credential: Credential<string | Record<string, unknown>>;
```

Another problem here is that the `Credential` type has some properties non-optional, and we need these fields to be optional if I understand this functionality correctly. If I added validation that the request contains fields with obligatory path values (`credentialSubject`, `type`), we can try to use this type, but I have already struggled with solving TS compilation error.

Reply:
Another thought is that if this service does not validate input JSON Schemas, the output is not guaranteed, so having that detailed response DTO can be misleading.

Reply:
Yes, compilation works when string is removed. However, I see that now response schema generated is empty for `credential` field. I am not sure if NestJS is able to generate it at all when using this type...

Reply:
> Another thought is that if this service does not validate input JSON Schemas, the output is not guaranteed, so having that detailed response DTO can be misleading.

Hmm, fair point. Maybe it would be good to use the `Partial` type (https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype) : `Partial<Credential<Record<string, unknown>>>`

> However, I see that now the response schema generated is empty for the credential field. I am not sure if NestJS is able to generate it at all when using this type.

Maybe it needs to be an additional DTO that implements the interface?
```suggestion
  credential: CredentialDto;
```
where `CredentialDto implements Partial<Credential<Record<string, unknown>>>`